src/operator/combineLatest.ts

```diff
@@ -94,15 +100,21 @@ export function combineLatestStatic<T, T2, T3>(v1: ObservableInput<T>, v2: Obser
+export function combineLatestStatic<T, T2, T3, T4, T5, T6, T7>(v1: ObservableInput<T>, v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, v6: ObservableInput<T6>, v7: ObservableInput<T7>, scheduler?: Scheduler): Observable<[T, T2, T3, T4, T5, T6, T7]>;
+export function combineLatestStatic<T, T2, T3, T4, T5, T6, T7, T8>(v1: ObservableInput<T>, v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, v6: ObservableInput<T6>, v7: ObservableInput<T7>, v8: ObservableInput<T8>, scheduler?: Scheduler): Observable<[T, T2, T3, T4, T5, T6, T7, T8]>;
+export function combineLatestStatic<T, T2, T3, T4, T5, T6, T7, T8, T9>(v1: ObservableInput<T>, v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, v6: ObservableInput<T6>, v7: ObservableInput<T7>, v8: ObservableInput<T8>, v9: ObservableInput<T9>, scheduler?: Scheduler): Observable<[T, T2, T3, T4, T5, T6, T7, T8, T9]>;
 export function combineLatestStatic<T, R>(v1: ObservableInput<T>, project: (v1: T) => R, scheduler?: Scheduler): Observable<R>;
 export function combineLatestStatic<T, T2, R>(v1: ObservableInput<T>, v2: ObservableInput<T2>, project: (v1: T, v2: T2) => R, scheduler?: Scheduler): Observable<R>;
 export function combineLatestStatic<T, T2, T3, R>(v1: ObservableInput<T>, v2: ObservableInput<T2>, v3: ObservableInput<T3>, project: (v1: T, v2: T2, v3: T3) => R, scheduler?: Scheduler): Observable<R>;
 export function combineLatestStatic<T, T2, T3, T4, R>(v1: ObservableInput<T>, v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, project: (v1: T, v2: T2, v3: T3, v4: T4) => R, scheduler?: Scheduler): Observable<R>;
 export function combineLatestStatic<T, T2, T3, T4, T5, R>(v1: ObservableInput<T>, v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, project: (v1: T, v2: T2, v3: T3, v4: T4, v5: T5) => R, scheduler?: Scheduler): Observable<R>;
 export function combineLatestStatic<T, T2, T3, T4, T5, T6, R>(v1: ObservableInput<T>, v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, v6: ObservableInput<T6>, project: (v1: T, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6) => R, scheduler?: Scheduler): Observable<R>;
-export function combineLatestStatic<R>(...observables: Array<ObservableInput<any> | ((...values: Array<any>) => R) | Scheduler>): Observable<R>;
-export function combineLatestStatic<R>(array: ObservableInput<any>[], scheduler?: Scheduler): Observable<R>;
-export function combineLatestStatic<R>(array: ObservableInput<any>[], project: (...values: Array<any>) => R, scheduler?: Scheduler): Observable<R>;
```

Comment:
@kwonoj 

> if there's any actual usecases of more than 6? @saneyuki , do you actually have usecases?

Yes. There are some actual usecases in the code base of my place of work (I cannot indicate them because they are not open sourced). 

Their code uses `combineLatest()` to develop a input form with Rx like following:

```javascript
const input1 = ...;
const input2 = ...;
const input3 = ...;
const input4 = ...;
const input5 = ...;
const input6 = ...;
const input7 = ...;

function combine(v1: T, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6, v7: T7): FormViewModel {
  return new FormViewModel(v1, v2, v3, v4, v5, v6, v7);
}

const form = Rx.Observable.combineLatest(input1, input2, input3, input4, input5, input6, input7, combine);
```

- If we need more discussion about to introduce ~`T9` type parameters, I remove their changes from this pull request and I'll open a new pull request to add only them.
- As a this library's design, we cannot introduce ~`T9` type parameters at this moment, I (and my co-workers and places of work) may be able to accept its decision.